musica/bindings/micm/micm.cpp

```diff
@@ -10,71 +10,37 @@
 
 #include <mechanism_configuration/v1/types.hpp>
 
+#include <iostream>
+
 namespace py = pybind11;
 namespace v1 = mechanism_configuration::v1::types;
 
-PYBIND11_MAKE_OPAQUE(std::vector<micm::Conditions>)
-
-void bind_musica(py::module_ &core)
+void bind_micm(py::module_ & micm)
 {
-  py::bind_vector<std::vector<micm::Conditions>>(core, "VectorConditions");
-
-  py::class_<micm::Conditions>(core, "_Conditions")
-      .def(py::init<>())
-      .def_readwrite("temperature", &micm::Conditions::temperature_)
-      .def_readwrite("pressure", &micm::Conditions::pressure_)
-      .def_readwrite("air_density", &micm::Conditions::air_density_);
-
-  py::class_<musica::State>(core, "_State")
+  py::class_<musica::MICM>(micm, "MICM")
       .def(py::init<>())
-      .def("__del__", [](musica::State &state) {})
-      .def("number_of_grid_cells", [](musica::State &state) { return state.NumberOfGridCells(); })
-      .def_property(
-          "conditions",
-          [](musica::State &state) -> std::vector<micm::Conditions> & { return state.GetConditions(); },
-          nullptr,
-          "list of conditions structs for each grid cell")
-      .def_property(
-          "concentrations",
-          [](musica::State &state) -> std::vector<double> & { return state.GetOrderedConcentrations(); },
-          nullptr,
-          "native 1D list of concentrations, ordered by species and grid cell according to matrix type")
-      .def_property(
-          "user_defined_rate_parameters",
-          [](musica::State &state) -> std::vector<double> & { return state.GetOrderedRateParameters(); },
-          nullptr,
-          "native 1D list of user-defined rate parameters, ordered by parameter and grid cell according to matrix type")
-      .def("concentration_strides", [](musica::State &state) { return state.GetConcentrationsStrides(); })
-      .def(
-          "user_defined_rate_parameter_strides",
-          [](musica::State &state) { return state.GetUserDefinedRateParametersStrides(); });
-
-  py::enum_<musica::MICMSolver>(core, "_SolverType")
-      .value("rosenbrock", musica::MICMSolver::Rosenbrock)
-      .value("rosenbrock_standard_order", musica::MICMSolver::RosenbrockStandardOrder)
-      .value("backward_euler", musica::MICMSolver::BackwardEuler)
-      .value("backward_euler_standard_order", musica::MICMSolver::BackwardEulerStandardOrder)
-      .value("cuda_rosenbrock", musica::MICMSolver::CudaRosenbrock);
-
-  py::class_<musica::MICM>(core, "_Solver");
-
-  core.def(
+      .def("__del__", [](musica::MICM* self) { delete self; });
```

Comment:
Oh, I think we wrote this wrong the first time. We should probably call [`DeleteMicm`](https://github.com/NCAR/musica/blob/82b8b59f68352043fa3920dbabe2257848463c4f/include/musica/micm/micm_c_interface.hpp#L47), maybe?